compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -178,7 +180,7 @@ class ExtractSemanticDB extends Phase:
                 if !excludeChildren(tree.symbol) then
                   traverseChildren(tree)
             }
-            if !excludeDef(tree.symbol) && tree.span.hasLength then
+            if !excludeDef(tree.symbol) && (tree.span.hasLength || tree.symbol.isAnonymousClass) then
```

Comment:
The change seems kind of monkey patch, maybe we should refactor `ExtractSemanticDB` at some point.

Reply:
A redesign would be welcome - The only specification I used in the original implementation was comparing with the expect files from scalameta's semanticdb tests, so everything grew organically around that